### PR TITLE
Enhancement: Add latest sonic changes

### PR DESCRIPTION
**Pull request by vshatravenko**
* Add Engine creation + market update changes

### CODE REVIEW

**Review by akhlopiachyi — APPROVED** [review]

File: views/index.html [hunk 1]
```diff
@@ -1,8 +1,22 @@
 {{define "head"}}
+    <head>
+        {{ if .cssFiles}}
+            {{range .cssFiles}}
+                <link href={{ . }} rel="stylesheet">
+            {{end}}
+        {{end}}
+    </head>
```
Comment (akushniruk) [inline views/index.html:8]: better to put it in master.html because we need apply CSS from baseapp to SSR UI components

File: views/index.html [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+
 {{define "content"}}
     <body>
-        {{ .data }}
+        <div id={{ .rootID }}></div>
+        {{ if .jsFiles}}
+            {{range .jsFiles}}
+                <script src="{{ . }}"></script>
+            {{end}}
+        {{end}}
```
Comment (akushniruk) [inline views/index.html:19]: should be in master.html. In the future, we want to apply react-hydrate

File: web/src/components/CombinedOrderBook/index.tsx [hunk 3]
```diff
@@ -51,37 +51,52 @@ export interface CombinedOrderBookProps {
...
-        scroll.scrollTop = scroll.scrollHeight;
-      }
+        const scroll = document.getElementsByClassName('cr-order-book')[0];
+
+        if (!this.props.isLarge && scroll) {
+          scroll.scrollTop = scroll.scrollHeight;
+        }
     }
 
     public componentWillReceiveProps(next: CombinedOrderBookProps) {
```
Comment (akushniruk) [inline web/src/components/CombinedOrderBook/index.tsx:78]: let's change it to componentDidUpdate

File: web/src/containers/OrderBook/index.tsx [hunk 4]
```diff
@@ -72,57 +74,107 @@ class OrderBookContainer extends React.Component<Props, State> {
...
+    public componentDidMount() {
+        window.addEventListener('resize', this.handleResize);
+        this.handleResize();
+    }
+
+    public componentDidUpdate(prevProps: Props) {
+        this.handleResize();
+
+        if (!prevProps.orderBookLoading && this.props.orderBookLoading) {
+            setTimeout(() => {
```
Comment (akushniruk) [inline web/src/containers/OrderBook/index.tsx:86]: For what we need set timeout in this place?

**Review by akushniruk — CHANGES_REQUESTED** [review]

**Review by akushniruk — APPROVED** [review]

**Pull request merged**